### PR TITLE
ddl: fix enable metadata lock

### LINKED ISSUE #41796: Can't open metadata lock
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
mysql> show global variables like "%tidb_enable_metadata_lock%";
+---------------------------+-------+
| Variable_name             | Value |
+---------------------------+-------+
| tidb_enable_metadata_lock | ON    |
+---------------------------+-------+
1 row in set (0.00 sec)

mysql> set global tidb_enable_metadata_lock=0;
Query OK, 0 rows affected (0.15 sec)

mysql> show global variables like "%tidb_enable_metadata_lock%";
+---------------------------+-------+
| Variable_name             | Value |
+---------------------------+-------+
| tidb_enable_metadata_lock | OFF   |
+---------------------------+-------+
1 row in set (0.00 sec)

mysql> set global tidb_enable_metadata_lock=1;
Query OK, 0 rows affected (0.02 sec)

mysql> show global variables like "%tidb_enable_metadata_lock%";
+---------------------------+-------+
| Variable_name             | Value |
+---------------------------+-------+
| tidb_enable_metadata_lock | OFF   |
+---------------------------+-------+
1 row in set (0.00 sec)
```

### 2. What did you expect to see? (Required)

```

mysql> set global tidb_enable_metadata_lock=1;
Query OK, 0 rows affected (0.02 sec)


mysql> show global variables like "%tidb_enable_metadata_lock%";
+---------------------------+-------+
| Variable_name             | Value |
+---------------------------+-------+
| tidb_enable_metadata_lock | ON    |
+---------------------------+-------+
1 row in set (0.00 sec)

```

### 3. What did you see instead (Required)

Is it the reason for the problem?

https://github.com/pingcap/tidb/blob/96e345d1b3b88e0fb46d93d5117f666e353c9cf3/ddl/ddl.go#L1247
```

	if enable {
		sql := fmt.Sprintf("UPDATE HIGH_PRIORITY %[1]s.%[2]s SET VARIABLE_VALUE = %[4]d WHERE VARIABLE_NAME = '%[3]s'",
			mysql.SystemDB, mysql.GlobalVariablesTable, variable.TiDBEnableMDL, 0)
		sess, err := d.sessPool.get()
		if err != nil {
			logutil.BgLogger().Warn("[ddl] get session failed", zap.Error(err))
			return nil
		}
		defer d.sessPool.put(sess)
		se := newSession(sess)
		_, err = se.execute(ctx, sql, "disableMDL")
		if err != nil {
			logutil.BgLogger().Warn("[ddl] disable MDL failed", zap.Error(err))
		}
		return nil
	}

```

### 4. What is your TiDB version? (Required)

```

mysql> SELECT tidb_version()\G
*************************** 1. row ***************************
tidb_version(): Release Version: v6.6.0
Edition: Community
Git Commit Hash: f4ca0821fb96a2bdd37d2fb97eb26c07fc58d4e4
Git Branch: heads/refs/tags/v6.6.0
UTC Build Time: 2023-02-17 14:49:02
GoVersion: go1.19.5
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: tikv
1 row in set (0.00 sec)

```

### PR DESCRIPTION
**Pull request by jiyfhust**
### What problem does this PR solve?

Issue Number: close #41796

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

**Review by wjhuang2016 — COMMENTED** [review]
Could you add a unit test?

File: ddl/ddl.go [hunk 1]
```diff
@@ -1241,31 +1241,14 @@ func (d *ddl) startCleanDeadTableLock() {
 
 // SwitchMDL enables MDL or disable DDL.
```
Comment (Benjamin2037) [inline ddl/ddl.go:1242]:
```suggestion
// SwitchMDL enables MDL or disable MDL.
```

**Review by tangenta — APPROVED** [review]

**Pull request merged**